buckets/buckets.go

```diff
@@ -136,6 +136,12 @@ func (b *Buckets) Create(ctx context.Context, dbID thread.ID, key, name string,
 		}
 		return b.Create(ctx, dbID, key, name, pth, opts...)
 	}
+	if isInvalidSchemaErr(err) {
+		if err := b.updateCollection(ctx, dbID, opts...); err != nil {
+			return nil, err
+		}
+		return b.Create(ctx, dbID, key, name, pth, opts...)
```

Comment:
Q: So if creation fails because of a schema mismatch, we'll automatically update the collection schema?

Reply:
Yep, too heavy handed? We might have to revisit in a multi-writer world, but here it seems easiest to just not involve the user in the migration process.

Reply:
Yeah, sounds good. The only risk is having an unwanted/error change this side, and changing the schema to an unwanted thing. But I guess that would be reversible, right?

Reply:
Yep, totally